Model/Service/Facet/BuildGraphQlFacetService.php

```diff
@@ -52,12 +84,49 @@ public function getFacets(array $requestData): Facet
         $includeFilters = new IncludeFilters();
         $excludeFilters = new ExcludeFilters();
 
-        if (isset($requestData['filters']['price_filter'])) {
-            $priceFilters = $requestData['filters']['price_filter'];
-            $includeFilters->setPrice(
-                isset($priceFilters['from']) ? $priceFilters['from'] : null,
-                isset($priceFilters['to']) ? $priceFilters['to'] : null
-            );
+        foreach ($requestData['filters'] as $filter) {
+            if ($filter['name'] === 'category_filter' || // Skip visibility and category filters
+                $filter['name'] === 'visibility_filter'
+            ) {
+                continue;
+            } elseif ($filter['name'] === 'price_filter') { // Price filters
+                $includeFilters->setPrice(
+                    isset($filter['from']) ? $filter['from'] : null,
+                    isset($filter['to']) ? $filter['to'] : null
+                );
+            } else { // Custom field filters
+                try {
+                    $attributeCode = $filter['field'];
+                    $filterValues = $filter['value'];
+                    $attribute = $this->productAttributeRepository->get($attributeCode);
+                    $values = [];
+
+                    if (is_string($filterValues)) { // eq attribute
```

Comment:
I'd also check for `''`, will `getOptionText` crash or return empty?

Reply:
Changed the whole logic in the new method. Also, filters are not even generated if the request is empty string.